### ISSUE TITLE
Error when generating pdf

### ISSUE THREAD

**Issue by lvmajor** [microsoft/playwright-sharp#884]
When trying to generate a pdf after updating to latest nuget package (0.111.2), I receive the following error:
`PlaywrightSharpException: options.format.toLowerCase is not a function`

after calling it like this:
```
var result = await page.GetPdfAsync(null, pdfConversionOptions.Scale, pdfConversionOptions.DisplayHeaderFooter, pdfConversionOptions.HeaderTemplate, pdfConversionOptions.FooterTemplate,
  pdfConversionOptions.PrintBackground, pdfConversionOptions.Landscape, pdfConversionOptions.PageRanges, pdfConversionOptions.Format, pdfConversionOptions.Width, pdfConversionOptions.Height,
  pdfConversionOptions.Margin, pdfConversionOptions.PreferCSSPageSize);
```

I tried also specifying directly `PaperFormat.Letter` for example but I receive the same error...

**Comment by kblok** [comment]
I'll take a look at it!

**Comment by lvmajor** [comment]
Thanks!
Meanwhile I was able to pass over it by simply specifying `null` for the `PaperFormat` parameter, but now the margins don't seem to be working at all, I'm currently investigating it hehe

**Comment by kblok** [comment]
What did you have on pdfConversionOptions.Format?

**Comment by lvmajor** [comment]
The pdfConversionOptions object is the following
```
var pdfConversionOptions = new PlaywrightSharpPdfOptions
{
    Format = PaperFormat.Letter,
    PrintBackground = true,
    DisplayHeaderFooter = true,
    HeaderTemplate = await _viewRenderer.RenderViewToStringAsync("~/wwwroot/ChromeHeadless/HeaderTemplate.cshtml"),
    FooterTemplate = await _viewRenderer.RenderViewToStringAsync($"~/wwwroot/ChromeHeadless/FooterTemplate-{langCode}.cshtml"),
    Margin = new Margin
    {
        Top = "0.8in",
        Bottom = "0.4in",
        Left = "0.30in",
        Right = "0.30in"
    },
    Scale = 1
};
```

**Comment by lvmajor** [comment]
Okay for the margins, it's a css problem I think, so no need to investigate on your side for that, but for the error, I still haven't figured it out

**Comment by lvmajor** [comment]
Hmmm... I must be doing something really wrong as even the simplest case does not seem to work (the margins are not even working with a simple repro, so I ruled out the css side in the end I think.....). I followed the demo in the demos folder and modified it a bit to return a file instead of saving it to disk, plus using header and footer templates, but otherwise it's pretty much the same as the demo.

Could you take a look at the [repro I made](https://github.com/set-webmaster/TestPlaywrightSharp0_111_2) and tell me if there's any obvious mistake? (It's a stripped down version of what I'm actually doing in my real project, but I'm thinking if I can't even make this one work, no chance I will be able to with my real project ahah :P )

Thanks again for your time!

**Comment by kblok** [comment]
Will do @lvmajor!

**Comment by kblok** [comment]
Could you guide me to the steps to repro in your code?

**Comment by lvmajor** [comment]
Yeah no problem simply run the app, I edited the home page index endpoint, so it is generating the pdf file by default (see HomeController.cs -> Index endpoint on line 23)

**Comment by lvmajor** [comment]
Small update, I was wondering if the example was representative or if it was causing an error due to something different than simply switching version of the PlaywrightSharp Nuget package so I did create a new [branch in the repository](https://github.com/set-webmaster/TestPlaywrightSharp0_111_2/tree/Previous-package-version) linked above in which I reverted to the old package with the same test scenario. 

I confirmed it does work as expected with the previous package (PlaywrightSharp.Chromium 0.10.3)

**Comment by lvmajor** [comment]
@kblok no problem!

Now I'd like to know something.... should I wait until the v1.4 package is released before continuing investigating?

I'm wondering if it is possible that it would be fixed in this expected release or if you think this is something that will need to be fixed post release of version 1.4 (which is expected soon according to what you said in earlier comments/issues)

**Comment by kblok** [comment]
I hope the same 😂  in your code or mine :p

**Comment by kblok** [comment]
@lvmajor fix is on the way.
FYI. We are encouraging this kind of code style:

```cs
var bytesResponse = await page.GetPdfAsync(
                displayHeaderFooter: true,
                headerTemplate: headerTemplate,
                footerTemplate: "<div>Test footer template</div>",
                margin: new Margin { Bottom = "1in", Top = "2in" },
                printBackground: true);
```

**Comment by lvmajor** [comment]
@kblok for the code style I understand this is the way to go with the removal of the options object. But as I told you in a previous issue in my real scenario, I did a wrapper around Playwright/Puppeteer/WkHtmlToPdf in which I pass the options object around, so this code style is not suitable until the final call to Playwright. 

In the repro I provided, I just used the options object so that I could reproduce as much as possible what I'm doing in the actual project I'm working on :) But I totally understand the new encouraged code style hehe

So will the fix be published along with the new v1.4 or will the current package be patched as soon as possible?

thanks again for your time!
